### PR TITLE
JSUI-2839 Add SameSite=Lax when setting Cookie

### PR DESCRIPTION
**Pull request by ThibodeauJF**
https://coveord.atlassian.net/browse/JSUI-2839

[Deploy](https://dashboard.heroku.com/pipelines/a3535101-5bbf-4a5b-a909-47fcf8c9f149)

### CODE REVIEW

File: src/utils/CookieUtils.ts [hunk 1]
```diff
@@ -16,7 +16,7 @@ export class Cookie {
     host = location.hostname;
     if (host.split('.').length === 1) {
       // no '.' in a domain - it's localhost or something similar
-      document.cookie = this.prefix + name + '=' + value + expires + '; path=/';
+      document.cookie = this.prefix + name + '=' + value + expires + '; SameSite=Lax; path=/';
```
Comment (samisayegh) [inline src/utils/CookieUtils.ts]: Rather than duplicating this line, it would be better to place it in a function and reuse it.

**Comment by olamothe** [comment]
What happens if the cookie already exists for a given client, then they upgrade with this new param.

I'm guessing it does, but would it properly override the old one with this new one ?

**Comment by ThibodeauJF** [comment]
@olamothe Yes it does

File: src/utils/CookieUtils.ts [hunk 2]
```diff
@@ -3,20 +3,13 @@ export class Cookie {
   private static prefix: string = 'coveo_';
 
   static set(name: string, value: string, expiration?: number) {
-    let domain: string, domainParts: string[], date, expires: string, host: string;
+    let domain: string;
```
Comment (btaillon) [inline src/utils/CookieUtils.ts]: Could this be moved inside the `else` block?

File: src/utils/CookieUtils.ts [hunk 3]
```diff
@@ -31,24 +24,37 @@ export class Cookie {
...
 
+  private static buildCookie(name: string, value: string, expiration?: number, domain?: string) {
+    const expires = expiration ? this.buildExpiresValue(expiration) : '';
+    const domainCookie = domain ? `; domain=${domain}` : '';
+    return `${this.prefix}${name}=${value}${expires}${domainCookie}; SameSite=Lax; path=/`;
+  }
+
+  private static buildExpiresValue(expiration: number) {
+    const date = new Date();
+    date.setTime(date.getTime() + expiration);
```
Comment (btaillon) [inline src/utils/CookieUtils.ts]: Would replacing this with the following work?
```ts
const date = new Date(Date.now() + expiration);
```

**Pull request merged**